src/pool.rs

```diff
@@ -494,55 +523,65 @@ impl ConnectionPool {
         guard[address.shard].remove(address);
     }
 
-    /// Check if a replica can serve traffic. If all replicas are banned,
-    /// we unban all of them. Better to try then not to.
-    pub fn is_banned(&self, address: &Address, role: Option<Role>) -> bool {
-        let replicas_available = match role {
-            Some(Role::Replica) => self.addresses[address.shard]
-                .iter()
-                .filter(|addr| addr.role == Role::Replica)
-                .count(),
-            None => self.addresses[address.shard].len(),
-            Some(Role::Primary) => return false, // Primary cannot be banned.
-        };
+    /// Check if address is banned
+    /// true if banned, false otherwise
+    pub fn is_banned(&self, address: &Address) -> bool {
+        let guard = self.banlist.read();
 
-        debug!("Available targets for {:?}: {}", role, replicas_available);
+        match guard[address.shard].get(address) {
+            Some(_) => true,
+            None => false,
+        }
+    }
 
-        let guard = self.banlist.read();
+    /// Determines if we can try to unban this server
+    pub async fn can_unban(&self, address: &Address) -> bool {
```

Comment:
I don't think this is the right name for this function. can_unban tells me that you're checking if you're allowed to unban the instance given some condition, and then you can decide whether to do so or not. In this implementation, you're unbanning the instance if you can in the same function. Maybe a better name could be is_unbanned?

Reply:
changed to try_unban